### DISCUSSION TITLE
Why I'm reciving 500 Internal server error when I try to delete a service.

### DISCUSSION THREAD

**Discussion by aymanhouadria** [microcks/microcks#929] [Q&A] [answered]
When I try to delete a service either via Api or UI, I get this  error: _**Service cannot be deleted (Http failure response for https://microcks-xxxxx.xxxx.xxxx/api/services/646f433fc484bb6a697e41da: 500 OK).**_I have spent days trying to find a solution but there is no way. 
![image](https://github.com/microcks/microcks/assets/61763166/f9ea0367-3848-492a-b573-a09dbcdfb753)

**Comment by lbroudoux** [comment]
Hi @aymanhouadria,

Obviously, this is not the expected behavior. Sorry for troubles!
Some pieces of stuff to look at to investigate further:
* Your Microcks version (as found in About dialog in the UI),
* The role of your current user (it should be `manager` or `admin`)
* Wether you enabled the repository multi-tenancy feature,
* Any logs/stack traces on the server-side? 500 error typically produces stack traces

Let us know and we'll try to help our best.

**Comment by aymanhouadria** [comment]
Thank's for the reply. This is the info i have so far: 
![image](https://github.com/microcks/microcks/assets/61763166/69cd72d8-7b0b-44d0-a13a-14b7270a0ad0)

As for the server-side logs, this is what it gives me:
**[Request processing failed: java.lang.NullPointerException: Cannot invoke "io.github.microcks.security.UserInfo.getRoles()" because "userInfo" is null] with root cause
java.lang.NullPointerException: Cannot invoke "io.github.microcks.security.UserInfo.getRoles()" because "userInfo" is null**

The delete request is the only one that fails

**Reply by lbroudoux** [reply]
Ok! Thank you.
Funny enough I just fell on this one 2 hours ago... `1.8.0-SNAPSHOT` or `nightly` image has gone under big changes regrading how we manage security internally. We were previously relying on Keycloak Adapters but have now changed to Spring Security ones that are more generic but looks more difficult to configure... I'm currently testing in-depth this part and should produce a fix very quickly.
Let you know ASAP.

**Reply by lbroudoux** [reply]
This is being tracked here: https://github.com/microcks/microcks/issues/934

**Reply by lbroudoux** [reply, marked as answer]
A fix has been committed, and a new `quay.io/microcks/microcks:nightly` should be ready in a few minutes. Be sure to update your local image and let me know if it solves your issue. On my side, the stack trace is different as I'm on a more fresh version but the root cause should be same.

**Reply by aymanhouadria** [reply]
Thank you very much. I will try it tomorrow in the morning and I will make you know. Appreciate the fast fix

**Reply by lbroudoux** [reply]
Hello! Any news on this one? Did you find some time for a quick test of the fix?
